src/chainstate/coordinator/tests.rs

```diff
@@ -2287,6 +2289,319 @@ fn test_stx_transfer_btc_ops() {
+        //                          \ _ B24
+        //                            ....
+        //                          \ _ B30 -> B31 -> ...
+        let parent = if ix == 0 {
+            BlockHeaderHash([0; 32])
+        } else if ix >= 22 && ix <= 30 {
+            stacks_blocks[20].1.header.block_hash()
+        } else {
+            stacks_blocks[ix - 1].1.header.block_hash()
+        };
+
+        let burnchain_tip = burnchain.get_canonical_chain_tip().unwrap();
+        let next_mock_header = BurnchainBlockHeader {
+            block_height: burnchain_tip.block_height + 1,
+            block_hash: BurnchainHeaderHash([0; 32]),
+            parent_block_hash: burnchain_tip.block_hash,
+            num_txs: 0,
+            timestamp: 1,
+        };
+
+        // TODO: just added pox_consts (was None before)
+        let b = get_burnchain(path, pox_consts.clone());
+
+        let (good_op, block) = if ix == 0 {
+            make_genesis_block_with_recipients(
+                &sort_db,
+                &mut chainstate,
+                &parent,
+                miner,
+                10000,
+                vrf_key,
+                ix as u32,
+                // next_block_recipients.as_ref(),
+                None,
+            )
+        } else {
+            make_stacks_block_with_recipients(
+                &sort_db,
+                &mut chainstate,
+                &b,
+                &parent,
+                burnchain_tip.block_height,
+                miner,
+                1000,
+                vrf_key,
+                ix as u32,
+                // next_block_recipients.as_ref(),
+                None, 
+            )
+        };
+
+        let expected_winner = good_op.txid();
+        let mut ops = vec![good_op];
+
+        if ix == 0 {
+            // add a pre-stx op
+            ops.push(BlockstackOperationType::PreStx(PreStxOp {
+                output: first_del.clone(),
+                txid: next_txid(),
+                vtxindex: 5,
+                block_height: 0,
+                burn_header_hash: BurnchainHeaderHash([0; 32]),
+            }));
+            ops.push(BlockstackOperationType::PreStx(PreStxOp {
+                output: second_del.clone(),
+                txid: next_txid(),
+                vtxindex: 6,
+                block_height: 0,
+                burn_header_hash: BurnchainHeaderHash([0; 32]),
+            }));
+        } else if ix == 10 {
+            ops.push(BlockstackOperationType::DelegateStx(DelegateStxOp {
+                sender: first_del.clone(),
+                delegate_to: recipient_addr.clone(),
+                reward_addr: None,
+                delegated_ustx: delegated_amt,
+                until_burn_height: None,
+                txid: next_txid(),
+                vtxindex: 5,
+                block_height: 0,
+                burn_header_hash: BurnchainHeaderHash([0; 32]),
+            }));
+        } else if ix == 20 {
```

Comment:
Note: `ix == 20` means that the on-BTC operation takes effect in the Stacks block mined at `ix == 21`.